src/js/ol3-map-widget.js

```diff
@@ -272,9 +272,51 @@
         iconFeature.setStyle(style);
     };
 
+    var format_builders = {
+        "GPX": ol.format.GPX,
+        "KML": ol.format.KML,
+        "OSMXML": ol.format.OSMXML,
+        "WFS": ol.format.WFS,
+        "WMSGetFeatureInfo": ol.format.WMSGetFeatureInfo,
+        "IGC": ol.format.IGC,
+        "Polyline": ol.format.Polyline,
+        "WKT": ol.format.WKT,
+        "MVT": ol.format.MVT,
+        "EsriJSON": ol.format.EsriJSON,
+        "GeoJSON": ol.format.GeoJSON,
+        "TopoJSON": ol.format.TopoJSON
+    }
+
+    var addFormat = function addFormat(layer_info) {
+        if (!layer_info.format) {
+            return undefined;
+        }
+
+        if (layer_info.format === "GML") {
+            return new ol.format.GML({srsName: layer_info.srsName});
+        }
+        if (layer_info.format === "GML2") {
+            return new ol.format.GML2({srsName: layer_info.srsName});
+        }
+        if (layer_info.format === "GML3") {
+            return new ol.format.GML3({srsName: layer_info.srsName});
+        }
+
+        return new format_builders[layer_info.format]()
+    }
+
     Widget.prototype.addLayer = function addLayer(layer_info) {
-        var layer, params, service_url;
+        var layer;
 
+        layer = layer_builders[layer_info.type](layer_info);
+
+        this.map.addLayer(layer);
+
+        this.layers[layer_info.name] = layer;
+    };
+
+    var addImageWMSLayer = function addImageWMSLayer(layer_info) {
+        var layer, params, service_url;
         params = {
```

Comment:
Missing support for other parameters. Documentation of the param option from openlayers:
> WMS request parameters. At least a LAYERS param is required. STYLES is '' by default. VERSION is 1.3.0 by default. WIDTH, HEIGHT, BBOX and CRS(SRS for WMS version < 1.3.0) will be set dynamically. Required.